[ONNX] Skip flaky dynamic tests before ORT==1.15 in fx exporter

titaiwangms opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98856

Disable all flaky dynamic tests
From https://github.com/pytorch/pytorch/issues/98626#issuecomment-1502692018

Rerun all test cases and update skip reasons. The cases failing on both static and dynamic shapes are unittest.skipped. If it only fails on dynamic, it's skipped by skip_dynamic_test. There are a few skipped with skip_ort_min_version, since ORT is not supporting dynamic fx exporter until next version.

titaiwangms committed:
[ONNX] Skip flaky dynamic tests before ORT==1.15 in fx exporter

[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Skip flaky dynamic tests before ORT==1.15 in fx exporter"


Disable all flaky dynamic tests
From https://github.com/pytorch/pytorch/issues/98626#issuecomment-1502692018

[ghstack-poisoned]

BowenBao commented on test/onnx/pytorch_test_common.py:
```diff
@@ -158,14 +158,19 @@ def wrapper(self, *args, **kwargs):
 
 
-def skip_min_ort_version(reason: str, version: str):
+# TODO(titaiwang): dynamic_only is pecific to the situation that dynamic fx exporter
```
```suggestion
# TODO(titaiwang): dynamic_only is specific to the situation that dynamic fx exporter
```

titaiwangms replied:
Done

BowenBao commented on test/onnx/test_fx_to_onnx_with_onnxruntime.py:
```diff
@@ -179,6 +190,9 @@ def tearDown(self):
         super().tearDown()
 
+    @pytorch_test_common.skip_min_ort_version(
+        reason="Flaky test", version="1.15", dynamic_only=True
```
Mention the reason why test is flaky

titaiwangms replied:
Done

BowenBao reviewed (approved):
LGTM, minor comments

titaiwangms committed:
Update on "[ONNX] Skip flaky dynamic tests before ORT==1.15 in fx exporter"


Disable all flaky dynamic tests
From https://github.com/pytorch/pytorch/issues/98626#issuecomment-1502692018

[ghstack-poisoned]

titaiwangms commented:
@pytorchbot merge
